docs/adr/0001-add-generic-fetcher.md

```diff
@@ -0,0 +1,109 @@
+# Add generic fetcher
+
+- Status: proposed
+- Date: 2024-10-30
+
+## Context
+
+The main motivation for this change is to cover use cases of users that need to download arbitrary files that don't fit
+within an established package ecosystem cachi2 could potentially otherwise support. The target audience is users that
+want to use cachi2 to achieve hermetic builds and want an easy way to also include these arbitrary files, that cachi2
+will account for in the SBOM it produces.
+
+## Decision
+
+This change introduces a generic fetcher, an additional cachi2 package manager. This package manager utilizes a custom
+lockfile that is located in the input repository. Based on that lockfile, it will download files, save them into a requested
+location, and verify checksums. Below is a more detailed overview of the implementation.
+
+### Lockfile format
+
+Cachi2 expects the lockfile to be named `generic_lockfile.yaml`.
```

Comment:
Nitpick: I don't quite like using underscore on filenames. I also think the name could be slightly clearer, such as `generic-artifacts.lock.yaml`.

Another suggestion would be to namespace it with `cachi2`, so that we make a clear statement that it is a Cachi2-defined file (something like `cachi2-artifacts.lock.yaml`). I know we had opinions against this in the original design, but even with the imminent Cachi2 name change, it might still make sense to namespace the lockfile name.

In the end, this is still a nitpick, I see no big problems with keeping the current name.

Reply:
I think it's not a good idea to namespace it with cachi2 for two reasons: 1) it's gonna change soon 2) it's inconsistent with the RPM package manager naming. 

Thefore I think it should be something similar as the [RPM lockfile name](https://github.com/containerbuildsystem/cachi2/blob/main/cachi2/core/package_managers/rpm/main.py#L30-L30), which is `rpms.lock.yaml`, like you mentioned `artifacts.lock.yaml`, or perhaps even more verbose `build-artifacts.lock.yaml`

Reply:
`artifacts.lock.yaml` is something I would have gone with. I was also thinking about namespacing it later on, because it does make sense to indicate that the format is custom to a particular tool, but we'd then have to support multiple names for the files which I don't want. If anything, we should allow users to pass these files via CLI and then it can be named literally anything, so what @kosciCZ proposes works for me.

Reply:
+1 for `artifacts.lock.yaml`

Reply:
Default  lockfile name is now `artifacts.lock.yaml`.